test/jdk/sun/security/pkcs11/Signature/SignatureTestPSS.java

```diff
@@ -62,82 +62,57 @@ public static void main(String[] args) throws Exception {
 
     @Override
     public void main(Provider p) throws Exception {
-        Signature sig;
-        try {
-            sig = Signature.getInstance(SIGALG, p);
-        } catch (NoSuchAlgorithmException e) {
+        if (!PSSUtil.isSignatureSupported(p)) {
             System.out.println("Skip testing RSASSA-PSS" +
                 " due to no support");
             return;
         }
-        this.prov = p;
-        for (int i : KEYSIZES) {
-            runTest(i);
-        }
-    }
-
-    private void runTest(int keySize) throws Exception {
-        byte[] data = new byte[100];
-        IntStream.range(0, data.length).forEach(j -> {
-            data[j] = (byte) j;
-        });
-        System.out.println("[KEYSIZE = " + keySize + "]");
 
-        // create a key pair
-        KeyPair kpair = generateKeys(KEYALG, keySize);
-        test(DIGESTS, kpair.getPrivate(), kpair.getPublic(), data);
-    }
-
-    private void test(String[] digestAlgs, PrivateKey privKey,
-            PublicKey pubKey, byte[] data) throws RuntimeException {
-        // For signature algorithm, create and verify a signature
-        for (String hash : digestAlgs) {
-            for (String mgfHash : digestAlgs) {
-                try {
-                    checkSignature(data, pubKey, privKey, hash, mgfHash);
-                } catch (NoSuchAlgorithmException | InvalidKeyException |
-                         SignatureException | NoSuchProviderException ex) {
-                    throw new RuntimeException(ex);
-                } catch (InvalidAlgorithmParameterException ex2) {
-                    System.out.println("Skip test due to " + ex2);
+        for (int kSize : KEYSIZES) {
+            System.out.println("[KEYSIZE = " + kSize + "]");
+            KeyPair kp = PSSUtil.generateKeys(p, kSize);
+            PrivateKey privKey = kp.getPrivate();
+            PublicKey pubKey = kp.getPublic();
+            for (String hash : DIGESTS) {
+                for (String mgfHash : DIGESTS) {
+                    System.out.println("    [Hash  = " + hash +
+                            ", MGF1 Hash = " + mgfHash + "]");
+                    PSSUtil.AlgoSupport s =
+                            PSSUtil.isHashSupported(p, hash, mgfHash);
+                    if (s == PSSUtil.AlgoSupport.NO) {
+                        System.out.println("    => Skip; no support");
```

Comment:
Similar request here to mark test as skipped if all algorithms are not supported.

Reply:
This is just one digest algorithm. To mark the test skipped if none of the algorithms are supported will require different handling.

Reply:
Yes est should only throw `SkippedEXception` when all algos are skipped.

Reply:
Re-factored test to throw SkippedException if all algos are skipped.